source/server/overload_manager_impl.h

```diff
@@ -36,25 +47,36 @@ class OverloadAction {
   // Returns the current action state, which is the max state across all registered triggers.
   OverloadActionState getState() const;
 
-  class Trigger {
-  public:
-    virtual ~Trigger() = default;
-
-    // Updates the current value of the metric and returns whether the trigger has changed state.
-    virtual bool updateValue(double value) PURE;
-
-    // Returns the action state for the trigger.
-    virtual OverloadActionState actionState() const PURE;
-  };
-  using TriggerPtr = std::unique_ptr<Trigger>;
-
 private:
+  using TriggerPtr = std::unique_ptr<Trigger>;
   absl::node_hash_map<std::string, TriggerPtr> triggers_;
   OverloadActionState state_;
   Stats::Gauge& active_gauge_;
   Stats::Gauge& scale_percent_gauge_;
 };
 
+class LoadShedPointImpl : public LoadShedPoint {
+public:
+  LoadShedPointImpl(const envoy::config::overload::v3::LoadShedPoint& config,
+                    Stats::Scope& stats_scope, Random::RandomGenerator& random_generator);
+  LoadShedPointImpl(const LoadShedPointImpl&) = delete;
+  LoadShedPointImpl& operator=(const LoadShedPointImpl&) = delete;
+
+  bool shouldShedLoad() override;
+  void updateResource(const std::string& name, double pressure);
```

Comment:
nit: Please comment since this is not part of the LoadShedPoint interface. In particular, what does `pressure` mean. Also probably absl::string_view instead of const string&.

Reply:
Good idea, done.